src/main/java/com/unstoppabledomains/resolution/contracts/BaseContract.java

```diff
@@ -99,6 +99,13 @@ private Tuple fetchMethod(String method, Object[] args) throws NamingServiceExce
       final String replacedAnswer = answer.replace("0x", "");
       return function.decodeReturn(FastHex.decode(replacedAnswer));
     } catch(IOException exception) {
+      if (exception.getMessage().matches("(.*)response code: (403|401|429)(.*)")) {
```

Comment:
What about 5xx?

Reply:
these are the error code that will make partners to be aware of when integrate with us such as authorized and rate limited. The others like 5xx could be thrown by RPC providers as well so let them handle it themselves.